apiserver/facades/client/client/status_test.go

```diff
@@ -101,6 +101,86 @@ func (s *statusSuite) TestFullStatusUnitLeadership(c *gc.C) {
 	c.Assert(unit.Leader, jc.IsTrue)
 }
 
+func (s *statusSuite) TestFullStatusInterfaceScaling(c *gc.C) {
+	machine := s.addMachine(c)
+	s.createSpaceAndSubnetWithProviderID(c, "public", "10.0.0.0/24", "prov-0000")
+	s.createSpaceAndSubnetWithProviderID(c, "private", "10.20.0.0/24", "prov-ffff")
+	s.createSpaceAndSubnetWithProviderID(c, "dmz", "10.30.0.0/24", "prov-abcd")
+
+	s.WaitForModelWatchersIdle(c, s.State.ModelUUID())
+	tracker := s.State.TrackQueries()
+
+	client := s.APIState.Client()
+	_, err := client.Status(nil)
+	c.Assert(err, jc.ErrorIsNil)
+
+	queryCount := tracker.ReadCount()
+
+	// Add a bunch of interfaces to the machine.
+	s.createNICWithIP(c, machine, "eth0", "10.0.0.11/24")
+	s.createNICWithIP(c, machine, "eth1", "10.20.0.42/24")
+	s.createNICWithIP(c, machine, "eth2", "10.30.0.99/24")
+
+	err = machine.SetDevicesAddresses(
+		state.LinkLayerDeviceAddress{
+			DeviceName:        "eth1",
+			ConfigMethod:      state.StaticAddress,
+			ProviderNetworkID: "vpc-abcd",
+			ProviderSubnetID:  "prov-ffff",
+			CIDRAddress:       "10.20.0.42/24",
+		},
+		state.LinkLayerDeviceAddress{
+			DeviceName:        "eth2",
+			ConfigMethod:      state.StaticAddress,
+			ProviderNetworkID: "vpc-abcd",
+			ProviderSubnetID:  "prov-abcd",
+			CIDRAddress:       "10.30.0.99/24",
+		},
+	)
+	c.Assert(err, jc.ErrorIsNil)
+
+	s.WaitForModelWatchersIdle(c, s.State.ModelUUID())
+	tracker.Reset()
+
+	_, err = client.Status(nil)
+	c.Assert(err, jc.ErrorIsNil)
+
+	// The number of queries should be the same.
+	c.Check(tracker.ReadCount(), gc.Equals, queryCount)
```

Comment:
Would it be worth using gc.Commentf to give context about what queries were done in the case that queryCount doesn't match? Ideally you would have a delta of the previous queries vs the current queries. I guess you can do that when you actually need to debug the failure.